src/http/RegisteredHeaders.h

```diff
@@ -57,7 +57,7 @@ enum HdrType {
     FORWARDED,                      /**< RFC 7239 */
     FROM,                           /**< RFC 7231 */
     HOST,                           /**< RFC 7230 */
-    HTTP2_SETTINGS,                 /**< RFC 7540 */
+    HTTP2_SETTINGS,                 /**< obsolete RFC 7540 header we may need to erase */
```

Comment:
You misunderstand the comment.  Erasure is up to admin configuration.

This and other headers are left as defined so that Squid can locate them quickly by-ID when admin want to do that erasure instead of performing a slow series of string comparisons on every request.

Reply:
Well, we can fix the comment so that others will not "misunderstand" it like I did, but those details are not the point of this change request. It seems wrong to optimize for the obsolete and rare header -- keeping this line slows down every Squid out there (by a tiny little bit) in hope to speed up a hypothetical use case that the RFC claims is very rare and ought to disappear completely. This is a "death by a thousand cuts" approach. We should be optimizing the _common_ case instead (by removing this obsolete registration).

> and the Connection header auto-erase to work efficiently

Registration has no effect on that AFAICT: HttpHeader::removeConnectionHeaderEntries() does not use registered header IDs.

Reply:
Done in [dae0ca8](https://github.com/squid-cache/squid/pull/1071/commits/dae0ca8059224fe4c108114e7fdbcc4d40552be0)